compiler/jsgen.nim

```diff
@@ -179,9 +179,24 @@ proc mapType(p: PProc; typ: PType): TJSTypeKind =
   else: result = mapType(typ)
 
 proc mangleName(s: PSym; target: TTarget): Rope =
+  proc validJsName(name: string): bool =
+    result = true
+    const reservedWords = ["break", "case", "catch", "class", "const", "continue",
```

Comment:
Ok. The lookup speed can definitely be improved by using sorted array + binary search.